crates/uv-cli/src/lib.rs

```diff
@@ -3135,6 +3135,21 @@ pub struct SyncArgs {
     #[arg(long, conflicts_with = "all_packages")]
     pub package: Option<PackageName>,
 
+    /// Sync the virtual environment for the specified PEP 723 Python script, rather than the current
+    /// project.
+    ///
+    /// If provided, uv will sync the dependencies based on the script's inline metadata table, in
+    /// adherence with PEP 723.
+    #[arg(
+        long,
+        conflicts_with = "active",
```

Comment:
Does it make sense to allow active e.g. for `uv sync --script --active scripts/publish/test_publish.py`? My use case is getting IDE support without explicit uv IDE support, having a shortcut for `uv export --script scripts/publish/test_publish.py | uv pip install -`. (I remember we had a discussion about that, but i can't find it)

Reply:
I sort of defer to @zanieb here because it _could_ cause complications if we ever want to allow "run a script in the context of a project".

Reply:
> "run a script in the context of a project".

Can you describe what this means briefly? Layered on top of the project environment?

I think `--active` makes sense here. I presume it's awkward to support given how I implemented `--active` though?

Reply:
I think `--active` wouldn't be too bad to support. I thought there was some thought that we might want you to be able to "run a script in the context of a project" -- I don't know what it means exactly, but there's some vague allusions to it in the code. Maybe, like, layering the script's dependencies atop the project's dependencies (like you said)?

Reply:
Did this separately: https://github.com/astral-sh/uv/pull/11433